### ISSUE TITLE
Decode additional strings.

### ISSUE THREAD

**Issue by MariasStory** [radare/radare2#11085]
Please decode strings from some simple obfuscation:
![string1](https://user-images.githubusercontent.com/6576335/44095047-134280b8-9fd8-11e8-98c0-ad19ff6c95cf.png)

It might be already implemented, but it would be nice to have the string in the strings list ;-)

**Comment by radare** [comment]
Use esil, anal.strings and theres another one that statically analyzes stack movements to find strings constructed like this but i dont remember ir right now.

and no, you wont see this in the strings list, because the string list you get is from RBin, aka by parsing the binary headrs, not by emulating the program, you’ll find a flag

**Comment by XVilka** [comment]
It make sense though to create something like `emu.str.*` namespace where to put the found strings after emulation.

**Comment by radare** [comment]
Flagspace?

**Comment by XVilka** [comment]
Yes, flagspace, where to push the flags for strings found during emulation, lets say during `aaaa` command. It might be very imprecise and error-prone, but certainly useful in some cases.

**Comment by radare** [comment]
Anal.strings do that but its not imprecise

**Comment by MariasStory** [comment]
Someone wants to try his skills on the this shellcode?
[download.zip](https://github.com/radare/radare2/files/2293909/download.zip)
You know the pass ;-)
I am sure that you can find out where it comes from.

Would be nice to see your results. This one is related to the first screenshot that I posted.

**Comment by MariasStory** [comment]
Hi team. Did someone check the shellcode? If so, did you get the strings extracted?
There are some more strings that also can be extracted but not seen in the disassembly.

**Comment by radare** [comment]
if you seek at the begining of the movs, enable emu.str=1 and do aepc $$ and aeim in there, the string will appear while you step in.. but its’ not automatically resolved with anal.strings because it doesnt assume the strings are constructred in the stack. so the issue is not solved, but you can write a script to automate that solution

**Comment by radare** [comment]
<img width="1440" alt="screen shot 2018-09-27 at 15 12 40" src="https://user-images.githubusercontent.com/917142/46177481-cc4fc500-c267-11e8-9e82-baf461f8d39b.png">

**Comment by MariasStory** [comment]
Hi @radare, Thanks for your interest in this topic. I have another example where an executable is reconstructed in memo using the "mov" obfuscation. It would be nice if r2 would extract the stuff without execution.
It is possible/planned?

**Comment by radare** [comment]
Can you share it?

**Comment by MariasStory** [comment]
Sure.
Here you go.
[shellDll_movJMP.zip](https://github.com/radare/radare2/files/2571268/shellDll_movJMP.zip)

**Comment by MariasStory** [comment]
Hi, did someone improve such strings detection/extraction?
I have more samples to play with ;-)

**Comment by radare** [comment]
can you give us some pointers about what you want exactly?

**Comment by MariasStory** [comment]
Hi Radare,

It would be nice to have something similar to the flare-floss - "FireEye Labs Obfuscated String Solver".
Instead the strings could appear in r2 analysis and in extracted strings list (like zzz) of radare2.
I know floss can export .r2 file and it kind of works.
I think that this functionality is better to be implemented in r2 and used in the native environment.
Although, if you think that it is not worth the effort, you can close the issue and I will use the floss.

Thanks,
Tolik

**Comment by radare** [comment]
Yes i think this feature makes sense to have it in r2 directly. Using anal.strings and aae it may resolve the strings referenced via esil. So i guess that enabling this and io.cache will be enough to get all the strings with f str.<tab>

Which command/action do u think it makes more sense to do this?

**Comment by radare** [comment]
We can also just extract strings out of the immediates in the instructions, but sometimes this is scrambled or xored before written into stack

**Comment by MariasStory** [comment]
Thanks, for following this idea.
I suggest to have approach similar to z vs zz vs zzz.
I mean that there should be option to have all strings logged/shown in r2 interface (in comments) or less noisy options for only stack related strings or strings that are kind of sorted out.
From forensic point of view, every string that was reconstructed/used/manipulated in binary can give a hint for finding the evidence (let's say in memory).
Possibly, I would emulate the (all) code with esil (maybe with some additional option) and get all possible strings that are generated/modified in this binary.
I know that you implemented part of it in the visible field. Maybe it can be extended to have it as a part of complete analysis results?
Also, it would be nice to have it in the way that is easy to use. Let's say you get additional results if you use more that 3z.
